drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -7112,7 +7231,10 @@ RenderingDevice::DrawListID RenderingDeviceVulkan::draw_list_begin(RID p_framebu
 		int color_count = 0;
 		for (int i = 0; i < framebuffer->texture_ids.size(); i++) {
 			Texture *texture = texture_owner.get_or_null(framebuffer->texture_ids[i]);
-			if (!texture || !(texture->usage_flags & TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)) {
+			// We only check for our VRS usage bit if this is not the first texture id.
+			// If it is the first we're likely populating our VRS texture.
+			// Bit dirty but..
+			if (!texture || (!(texture->usage_flags & TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT) && !(i != 0 && texture->usage_flags & TEXTURE_USAGE_VRS_ATTACHMENT_BIT))) {
```

Comment:
This is the other place I had an issue where in one scenario we're writing to the VRS image to set it up, in this case we do count it as an attachment that needs a clear color, in the other scenario we're using the VRS image as the VRS attachment in which case we definitely don't clear it.

Here again I'm assuming that if it is the first attachment, we must be writing to it asif its a normal image, if its any other attachment we're using it as the VRS attachment.

@reduz if you have any better suggestions, I'm all ears.